### ISSUE TITLE
Websocket connection stalls forever when running in dev

### ISSUE THREAD

**Issue by vgarleanu** [Dusk-Labs/dim#378]
It seems that commit 569f3753cc688465b8b7cdd27e98b478bd3c36f7 has introduced a bug where connecting to the websocket stalls forever and the "connecting to server" window never goes away. The connection eventually errors out after 2 minutes (timeout) or in rare cases gets through.

![Error](https://user-images.githubusercontent.com/20709310/150641357-d1ba47b3-8045-4b69-9180-929aecd71876.png)

The network tab shows that the connection is just pending
![image](https://user-images.githubusercontent.com/20709310/150641506-d885da92-d088-4c6d-a76d-b9592be20dbf.png)

The commit before (61c94b81d4efbb81dc18827d02cb5e8989eff090) seems to work just fine though.

Steps to reproduce:
* checkout into 569f3753cc688465b8b7cdd27e98b478bd3c36f7 (or master)
* run `yarn start`

**Comment by vgarleanu** [comment]
pinging @jimmycuadra

**Comment by vgarleanu** [comment]
Looking at the previous code I think this note might have something to do with this issue https://github.com/Dusk-Labs/dim/commit/2e2488580533363dc8eb8b5021b21455d098a148#diff-bfa11512f71e03fdea8299d7c34db7cda235da6c65d3111856a86fa00d894898L13
I remember that the http-proxy didnt proxy traffic well and it doesnt seem like the `proxy` field in the `package.json` does anything for websocket traffic.

**Comment by jimmycuadra** [comment]
That's odd. I've been using this version of the WebSockets code for a few weeks without problems. It does seem like Create React App's proxy handling is the cause though. I'll open a PR to use the previous approach with a hardcoded port based on NODE_ENV, but leave the other WebSocket changes as is.

**Comment by jimmycuadra** [comment]
I just noticed when looking at the diff in #379 that we're setting up our own proxy manually. I never noticed this before. I bet it conflicts with the proxy that Create React App sets up and that was the cause of this issue. It might also explain why I had to change from using "localhost" to using "127.0.0.1" in #369. A good follow up for this would be trying to remove our custom proxy and seeing if CRA's will work by itself.
